app/src/lib/utils/filters.ts

```diff
@@ -1,4 +1,28 @@
+import { isStr } from '$lib/utils/string';
+
 // If a value occurs > 1 times then all but one will fail this condition.
 export function unique(value: any, index: number, array: any[]) {
 	return array.indexOf(value) === index;
 }
+
+/**
+ * Filters an array of objects based on a specific string value in a given key.
+ *
+ * @template T - The type of objects in the array.
+ * @template K - The key of the object to filter by.
+ * @param {T[]} arr - The array of objects to filter.
+ * @param {K} key - The key to filter by.
+ * @param {string} value - The string value to search for in the specified key.
+ * @returns {T[]} - The filtered array of objects.
```

Comment:
If you agree with the other comment about jsdc then we can probably omit this too?

Reply:
Ah yeah, sorry should have thought about that one. Will take it out too